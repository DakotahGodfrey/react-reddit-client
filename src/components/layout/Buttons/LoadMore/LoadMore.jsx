import React from "react";
const LoadMore = ({ handleClick }) => {
  return <button onClick={handleClick}>Load More</button>;
};

export default LoadMore;
