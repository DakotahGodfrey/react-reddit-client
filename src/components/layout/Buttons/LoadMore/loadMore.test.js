import { screen, render, fireEvent } from "@testing-library/react";
import LoadMore from "./LoadMore";

describe("Load More", () => {
  const handleLoadMoreClick = jest.fn();
  it("should match the snapshot", () => {
    const wrapper = render(
      <LoadMore handleLoadMoreClick={handleLoadMoreClick} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  beforeEach(() => {
    render(<LoadMore handleLoadMoreClick={handleLoadMoreClick} />);
  });
  it("should render an accesible loadMore button element without error", () => {
    expect(screen.getByRole("button")).toBeInTheDocument();
    expect(screen.getByRole("button")).toHaveClass("load-more");
    expect(screen.getByRole("button")).toHaveTextContent("Load More");
    expect(screen.getByRole("button")).toHaveAttribute("aria-label");
  });
  it("should call the handleLoadMoreClick function when clicked", () => {
    fireEvent.click(screen.getByRole("button"));
    expect(handleLoadMoreClick).toHaveBeenCalledTimes(1);
  });
  it("should render an expand more icon, that is hidden from screen readers", () => {
    expect(screen.getByText("expand_more")).toHaveClass("material-icons");
    expect(screen.getByText("expand_more")).toHaveAttribute(
      "aria-hidden",
      "true"
    );
  });
});
