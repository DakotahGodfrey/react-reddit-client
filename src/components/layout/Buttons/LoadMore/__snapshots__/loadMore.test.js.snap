// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Load More should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-label="load more"
        class="load-more"
      >
        Load More
        <i
          aria-hidden="true"
          class="material-icons"
        >
          expand_more
        </i>
      </button>
    </div>
    <div>
      <button
        aria-label="load more"
        class="load-more"
      >
        Load More
        <i
          aria-hidden="true"
          class="material-icons"
        >
          expand_more
        </i>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      aria-label="load more"
      class="load-more"
    >
      Load More
      <i
        aria-hidden="true"
        class="material-icons"
      >
        expand_more
      </i>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
